coderedcms/models/page_models.py

```diff
@@ -455,6 +477,59 @@ def get_index_children(self):
 
         return query
 
+    @classmethod
+    def get_related_query_model(cls) -> Type[models.Model]:
+        """
+        Returns the model used for finding related content. By default this will be
+        siblings, and so will return itself.
+        """
+        if hasattr(cls, "related_querymodel"):
+            if isinstance(cls.related_query_model, Union[str, models.Model]):
+                return resolve_model_string(
+                    cls.related_querymodel, cls._meta.app_label
+                )
+            else:
+                raise AttributeError(
+                    (
+                        f"The related_querymodel should be a model or str."
+                        f" The related_querymodel of {cls} is {type(cls.related_querymodel)}"
+                    )
+                )
+        return cls
+
+    def get_related_pages(self) -> models.QuerySet:
+        """
+        Returns a queryset of the model type defined by realted_querymodel, defaulting to
+        self. Ordered by number of shared classifier terms, and then optionally by the
+        index_order_by.
+        """
+
+        order_by = []
+        # Get our related query model, and queryset. Exclude self, in case it is present
+        related_querymodel = self.get_related_query_model()
+        r_qs = related_querymodel.objects.all().live().exclude(id=self.id)
+
+        # if we have a preferred classifier term, order by that
+        if self.preferred_related_classifier_term:
+            p_ct_q = models.Q(
+                classifier_terms=self.preferred_related_classifier_term
+            )
+            r_qs = r_qs.annotate(p_ct=p_ct_q)
+            order_by.append("-p_ct")
+
+        # order by number of shared classifier terms
+        r_ct_q = models.Q(classifier_terms__in=self.classifier_terms.all())
+        r_qs = r_qs.annotate(r_ct=models.Count(r_ct_q))
+        order_by.append("-r_ct")
+
+        # if we include an index_order_by, order the set based on that
+        if self.index_order_by:
```

Comment:
So, this will probably not work as expected. For example, if the current class is an ArticlePage (likely scenario)... it will not be using `index_order_by`, that is set on the parent ArticleIndexPage.

Perhaps a slightly more effective strategy... keep the related_querymodel, and if it is set run a query on all of those as you are currently doing. But if it is NOT SET, then simply get this page's parent, and call its `get_index_children()` (which already does all the ordering etc. for you).